ara/callback/log_ara.py

```diff
@@ -85,13 +87,39 @@ def __init__(self):
 
     def get_or_create_host(self, hostname):
         try:
-            host = models.Host.query.filter_by(name=hostname).one()
+            host = (models.Host.query
+                    .filter_by(name=hostname)
+                    .filter_by(playbook_id=self.playbook.id)
+                    .one())
         except models.NoResultFound:
-            host = models.Host(name=hostname)
+            host = models.Host(name=hostname, playbook=self.playbook)
             db.session.add(host)
 
         return host
 
+    def get_or_create_file(self, path):
+        try:
+            file_ = (models.File.query
+                     .filter_by(path=path)
+                     .filter_by(playbook_id=self.playbook.id)
+                     .one())
+        except models.NoResultFound:
+            file_ = models.File(path=path, playbook=self.playbook)
+            db.session.add(file_)
+
+            # LKS: should collecting content be optional?
+            with open(path, 'r') as fd:
+                sha1 = hashlib.sha1(fd.read()).hexdigest()
```

Comment:
I don't have a strong opinion. If we feel we're going to be using the sha1 call in too many places, it should go in utils. This is just two places for the time being (model and here) but will it be used elsewhere as well ?